### ISSUE TITLE
Variable Type Confusion

### ISSUE THREAD

**Issue by bhsmither** [cubecart/v6#2980]
PHP 8.1 will issue a deprecation warning (removed in PHP 9) where an instance of:
```
$var = false;
$var['x'] = "value";
```
saying "Automatic conversion of false to array is deprecated."

The instance found is in `file.class.php`, near line 223. Tracing the code suggests:
In `cache.class.php`, there is `protected $_empties = array();`. This is good.
However, a call to `_getEmpties()` will call `read()`.
In `read()`, several situations will return `false`. This is bad for `_getEmpties()`.

Suggest that, in this instance, `_getEmpties()` returns an empty array when receiving a `false` from `read()`.

(More testing required.)

**Comment by bhsmither** [comment]
Also in `SEO->_getCategoryList()`, where `$this->_cat_dirs` receives `false` from `Cache->read()` but later wants to be an array.

Suggest initialize `SEO->_cat_dirs` to an empty array.

**Comment by bhsmither** [comment]
And `GUI-listSkins()`.

**Comment by bhsmither** [comment]
Suggest initializing `Category->_category_translations` to an empty array.

**Comment by bhsmither** [comment]
Suggest initializing `Cart->basket` and `Cart->basket_data` to empty arrays.

Near line 893, there is:
```
$this->basket_data[$hash] = $product;
```

**Comment by abrookbanks** [comment]
This is all very well but CubeCart isn't going to work on PHP 8.1 at all. I don't see the point in making it 8.0 compatible with we have a huge block at 8.1.

**Comment by bhsmither** [comment]
Well, I am succeeding at making it work on PHP8.1 - fixing errors as they are reported. There are only minor code changes - so far.

Maybe you know something I don't? Maybe discuss this on the forums?

I trust I can read "isn't going to work at all" as actually you **not** shuttering CubeCart when PHP 7.X goes EOL?

**Comment by bhsmither** [comment]
Also `ACP->showNavigation()`, the cache read returns `false`, but later `$navigation` wants to be an array.

**Comment by bhsmither** [comment]
Also `Hookloader->scan_all_plugins()`, `$plugins` is initialized `false` but later wants to be an array.

**Comment by bhsmither** [comment]
See: https://github.com/cubecart/v6/commit/da5391934952d2220630d10b5698cfe74cd4d2a9
for other variables that got initialized to `false` - that may should have been initialized to an empty array.

**Comment by abrookbanks** [comment]
I believe this is resolved as part of #2708
